python/ray/experimental/compiled_dag_ref.py

```diff
@@ -37,10 +37,9 @@ def __init__(
 
     def __str__(self):
         return (
-            f"CompiledDAGRef(_dag={self._dag}, "
-            f"_execution_index={self._execution_index}, "
-            f"_ray_get_called={self._ray_get_called}, "
-            f"_dag_output_channels={self._dag_output_channels})"
+            f"CompiledDAGRef(dag_output_channels={self._dag_output_channels}, "
```

Comment:
Can we follow the ObjectRef syntax here? I.e. no `dag_output_channels` string and just print the IDs directly. The issue with `dag_output_channels` is that I don't think this means anything to the user (DAG users shouldn't need to know about the channel concept).

We should also remove `ray_get_called` - unlike execution_index, this field is not uniquely identifying so I don't think we should print it as part of __str__. You could move this to a `debug_info() -> str` method if you want.

Reply:
Yeah let's discuss a bit more offline: we need to consider the special case of `CompositeChannel`, although it is not expected to be dag_output_channel.